util.c

```diff
@@ -792,3 +792,16 @@ b32_pton_extended_hex(const char* src, size_t src_sz,
 
 #endif /* ! HAVE_B32_PTON */
 
+void
+ldns_swap_bytes(unsigned char *buf, size_t len)
```

Comment:
I would propose to implement endian swapping memcpy. I think all uses of this function first use memcpy, then do swap_bytes on the same data. Could it perhaps swap bytes into target buffer instead and use memcpy if that is not necessary? That way it would not require temp variable usage.

But I am not sure, shoulw it use instead ``BN_lebin2bn`` or ``BN_bin2bn`` ?